src/conditions/base/condition.ts

```diff
@@ -1,38 +1,9 @@
 import Joi, { ValidationError } from 'joi';
 
-export interface ConditionConfig {
-  COMPARATORS: string[];
-}
-
-export interface ReturnValueTestConfig {
-  index?: number;
-  comparator: string;
-  value: string | number;
-}
-
-export const ConditionConfig: ConditionConfig = {
-  COMPARATORS: ['==', '>', '<', '>=', '<=', '!='],
-};
-
-export const makeReturnValueTest =
-  (): Joi.ObjectSchema<ReturnValueTestConfig> =>
-    Joi.object({
-      index: Joi.number().optional(),
-      comparator: Joi.string()
-        .valid(...ConditionConfig.COMPARATORS)
-        .required(),
-      value: Joi.alternatives(Joi.string(), Joi.number()).required(),
-    });
-
-export interface ConditionInterface {
-  schema: Joi.ObjectSchema;
-  defaults: unknown;
-  toObj: () => Record<string, unknown>;
-}
-
-export class Condition implements ConditionInterface {
+export class Condition {
```

Comment:
Perhaps `Condition` should be an abstract class and instantiation of `Condition` should be relayed to some factory method.

Reply:
Would a factory method reduce the ability for devs to circumvent the validity of the condition json? You mentioned a way around it to me in a different issue/PR, could that surface area for misuse be reduced by this?

Reply:
I think not in the particular case we've discussed, but it would be reasonable to prevent developers from instantiating `Condition` directly and having them cast it to a different condition type. This is one avenue that could cause a misuse of the API.